test/steps/infinispan.go

```diff
@@ -22,43 +22,60 @@ import (
 	"github.com/cucumber/godog"
 	"github.com/cucumber/messages-go/v10"
 	"github.com/kiegroup/kogito-cloud-operator/test/framework"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+/*
+	DataTable for Infinispan example:
+	| username | developer |
+	| password | mypass    |
+ */
+
 const (
-	//DataTable first column
-	infinispanUsernameKey     = "username"
```

Comment:
@MarianMacik @radtriste I think that the situation with DataTable column names deserves deeper discussion.
The problem with sharing of those constants is that we create an artificial dependency between go files. It also reduces visibility of the columns - how to find out what are the possible values for column?

One option would be to add an example for each DataTable, showing every possible configuration which can be applied - as done few lines above. The possible downside could be that example and actual functionality can get out of sync.
Another option is to list all the column names as constants with specific unique identifier (as constants in  same package are shared) as shown in these removed lines.

I would personally prefer combination of both methods, but I am open to another options.

What do you think?

Reply:
So you suggest to export these constants to a separate file? I thought that `dataindex.go` will serve this purpose. In fact it now does what you meant by

> Another option is to list all the column names as constants with specific unique identifier (as constants in same package are shared) as shown in these removed lines.

I will just add KogitoApp DataTable example to the `utils.go` method as this mostly contains parsing functions for KogitoApp.

Or we can merge everything about parsing DataTable to one file and move also examples of DataTables there.

Reply:
We could have one datatable handler file like @MarianMacik suggested.
And I like having specific unique identifier which should be self documentation.
Adding a datatable example is also a good thing but we need to keep it up to date :)

Reply:
OK, so we agree on leaving the constants in one file?

Then I will rename utils to utils_datatable or just datatable for example and will merge everything about Data Table in this one file.

About example tables, I can probably leave there a comment with each constant which will describe to which resources the constant applies (KogitoApp, DataIndex, Infinispan etc.) as the name of columns overlap. And having 3 different constants just for the same name ("infinispan") is a little bit overkill, don't you think? I am not sure about this last point.

Reply:
> And having 3 different constants just for the same name ("infinispan") is a little bit overkill

What do you mean ?

Reply:
My idea is to handle the DataTable in the same steps file where it belongs, with all the constants to be there too.

For example the test/steps/infinispan.go 
The file contains definitions of steps. So I would keep DataTable handling there too, as from my perspective it is part of steps definition. Also it would contain all the constants defining DataTable columns, with proper identifier (as the columns are specific for Infinispan).

So, the test/steps/infinispan.go would contain:
- constants (infinispanUsernameKey, infinispanPasswordKey.... ,prefixed with infinispan to mark that these constants belong to Infinispan)
- DataTable example showing all possible columns and their usage
- DataTable parsing function for Infinispan step (as it is integral part of step definition)

Then, for example test/steps/kogitodataindex.go would contain:
- constants (dataIndexInfinispanUsernameKey, dataIndexInfinispanPasswordKey.... ,prefixed with dataIndex to mark that these constants belong to DataIndex)
- DataTable example showing all possible columns and their usage
- DataTable parsing function for DataIndex step

Moving all DataTable parsing functions to one file is an option too. I am just worry that it will become fragile, less readable and can have tendencies to become out of sync as DataTable will be on different place than step definition.

Reply:
> > And having 3 different constants just for the same name ("infinispan") is a little bit overkill
> 
> What do you mean ?

I meant what Karel is suggesting :) So for "user" we will have in the end 3 different constants as "user" is used in the first column for data index, kogito app and infinispan data tables. Just wanted to be sure I understood correctly as there will be duplication.